core/src/main/scala/spinal/core/FixedPoint.scala

```diff
@@ -247,6 +248,17 @@ abstract class XFix[T <: XFix[T, R], R <: BitVector with Num[R]](val maxExp: Int
     copy.asInstanceOf[this.type]
   }
 
+  def truncated(maxExp: ExpNumber, bitCount: BitCount): T = {
+    val t = fixFactory(maxExp, bitCount)
+    t := this.truncated.asInstanceOf[T]
+    t
+  }
+  def truncated(maxExp: ExpNumber, resolution: ExpNumber): T = {
+    val t = fixFactory(maxExp, resolution)
+    t := this.truncated.asInstanceOf[T]
+    t
+  }
```

Comment:
Off-topic: Here I think Scala really lacks the ability to implement a trait for a type after the type definition (AFAIK). It would make it possible to have something like:

```scala
trait FixRes
implement FixRes for BitCount
implement FixRes for ExpNumber
```

Then you can have only one function to define, document and implement; accepting the two types; and you do not duplicate logic until there is an actual need to access the content of this second argument (using a `case`).